javascript/modules/client/test/unit/client.test.ts

```diff
@@ -23,49 +30,174 @@ const contextParams: ContextParams = {
+    expect(clientDaoWhitelistVotingStatus).toEqual(true);
   });
   it("Should create a failing client", async () => {
     contextParams.web3Providers = web3endpoints.failing;
     const context = new Context(contextParams);
-    const client = new ClientDaoWhitelist(context);
+    const clientDaoERC20Voting = new ClientDaoERC20Voting(context);
+    const clientDaoWhitelistVoting = new ClientDaoWhitelistVoting(context);
 
-    expect(client).toBeInstanceOf(ClientDaoWhitelist);
-    expect(client.web3).toBeInstanceOf(JsonRpcProvider);
+    expect(clientDaoERC20Voting).toBeInstanceOf(ClientDaoERC20Voting);
+    expect(clientDaoWhitelistVoting).toBeInstanceOf(ClientDaoWhitelistVoting);
+
+    expect(clientDaoERC20Voting.web3).toBeInstanceOf(JsonRpcProvider);
+    expect(clientDaoWhitelistVoting.web3).toBeInstanceOf(JsonRpcProvider);
 
-    const status = await client.checkWeb3Status();
-    expect(status).toEqual(false);
+    const clientDaoERC20VotingStatus = await clientDaoERC20Voting.checkWeb3Status();
+    expect(clientDaoERC20VotingStatus).toEqual(false);
+
+    const clientDaoWhitelistVotingStatus = await clientDaoWhitelistVoting.checkWeb3Status();
+    expect(clientDaoWhitelistVotingStatus).toEqual(false);
   });
   it("Should create a client, fail and shift to a working endpoint", async () => {
     contextParams.web3Providers = web3endpoints.failing.concat(
-      web3endpoints.working,
+      web3endpoints.working
     );
     const context = new Context(contextParams);
-    const client = new ClientDaoWhitelist(context);
+    const clientDaoERC20Voting = new ClientDaoERC20Voting(context);
+    const clientDaoWhitelistVoting = new ClientDaoWhitelistVoting(context);
 
-    expect(client).toBeInstanceOf(ClientDaoWhitelist);
-    expect(client.web3).toBeInstanceOf(JsonRpcProvider);
+    expect(clientDaoERC20Voting).toBeInstanceOf(ClientDaoERC20Voting);
+    expect(clientDaoWhitelistVoting).toBeInstanceOf(ClientDaoWhitelistVoting);
 
-    await client
+    expect(clientDaoERC20Voting.web3).toBeInstanceOf(JsonRpcProvider);
+    expect(clientDaoWhitelistVoting.web3).toBeInstanceOf(JsonRpcProvider);
+
+    await clientDaoERC20Voting
+      .checkWeb3Status()
+      .then(isUp => {
+        expect(isUp).toEqual(false);
+        return clientDaoERC20Voting.shiftWeb3Node().checkWeb3Status();
+      })
+      .then(isUp => {
+        expect(isUp).toEqual(true);
+      });
+
+    await clientDaoWhitelistVoting
       .checkWeb3Status()
-      .then((isUp) => {
+      .then(isUp => {
         expect(isUp).toEqual(false);
-        return client.shiftWeb3Node().checkWeb3Status();
+        return clientDaoWhitelistVoting.shiftWeb3Node().checkWeb3Status();
       })
-      .then((isUp) => {
+      .then(isUp => {
         expect(isUp).toEqual(true);
       });
   });
+  it("Should create a ERC20VotingDAO locally", async () => {
+    contextParams.network = 31337;
```

Comment:
In testing, every test case should be a pure function not reusing data with any other one. Constants could be used from the local file, but nothing prevents another background task from accessing contextParams and updating a value to its global reference